### PR TITLE
samples: peripheral: radio_test: Refactoring sample code

### PR DESCRIPTION
**Pull request by maje-emb**
PR introduces: 
- Using Zephyr's API to generate a random pattern
- Remove dependency on hal/nrf_nvmc.h

### CODE REVIEW

**Review by dchat-nordic — APPROVED** [review]

**Review by grochu — APPROVED** [review]
Good update, thanks.

**Comment by maje-emb** [comment]
Aligned configurations for net core.

File: samples/peripheral/radio_test/src/radio_test.c [hunk 1]
```diff
@@ -329,18 +321,21 @@ static void generate_modulated_rf_packet(uint8_t mode,
...
-			tx_packet[i + 1] = rnd8();
-		} else if (pattern == TRANSMIT_PATTERN_11001100) {
-			tx_packet[i + 1] = 0xCC;
-		} else if (pattern == TRANSMIT_PATTERN_11110000) {
-			tx_packet[i + 1] = 0xF0;
-		} else {
-			/* Do nothing. */
-		}
+	switch (pattern) {
+	case (TRANSMIT_PATTERN_RANDOM):
```
Comment (kapi-no) [inline samples/peripheral/radio_test/src/radio_test.c:325]: I think parentheses are redundant (also for the remaining cases)

```suggestion
	case TRANSMIT_PATTERN_RANDOM:
```

**Review by kapi-no — APPROVED** [review]

File: samples/peripheral/radio_test/src/radio_test.c (hunk 1 above)
Comment (maje-emb) [reply samples/peripheral/radio_test/src/radio_test.c]: Yes, fixed.

**Comment by grochu** [comment]
> @grochu, @maje-emb, @KAGA164, can we change the Codeowner of this sample from me to someone else?
> 
> There is another PR from WiFi folks with changes to this sample. Can you take a look at it and review it?
> 
> #9136

Good point, I'll change it and will review the other PR.

File: samples/peripheral/radio_test/src/radio_cmd.c [hunk 2]
```diff
@@ -10,7 +10,10 @@
 #include <zephyr/init.h>
 #include <zephyr/shell/shell.h>
 #include <zephyr/types.h>
+
+#if NRF_POWER_HAS_DCDCEN_VDDH || NRF_POWER_HAS_DCDCEN
```
Comment (KAGA164) [inline samples/peripheral/radio_test/src/radio_cmd.c:14]: Is this needed here?

File: samples/peripheral/radio_test/src/radio_test.c [hunk 3]
```diff
@@ -7,15 +7,17 @@
...
-#include <hal/nrf_power.h>
-#include <hal/nrf_rng.h>
 #include <nrfx_timer.h>
 #include <zephyr/kernel.h>
+#include <zephyr/random/rand32.h>
 
 #include <hal/nrf_egu.h>
 #include <helpers/nrfx_gppi.h>
 
+#if NRF_POWER_HAS_DCDCEN_VDDH || NRF_POWER_HAS_DCDCEN
```
Comment (KAGA164) [inline samples/peripheral/radio_test/src/radio_test.c:17]: Is this needed here?

**Review by KAGA164 — CHANGES_REQUESTED** [review]

File: samples/peripheral/radio_test/src/radio_cmd.c (hunk 2 above)
Comment (maje-emb) [reply samples/peripheral/radio_test/src/radio_cmd.c]: Good point. It is not needed.

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**